pkg/agent/controller/noderoute/node_route_controller_test.go

```diff
@@ -228,3 +232,165 @@ func TestIPInPodSubnets(t *testing.T) {
 	assert.Equal(t, false, c.Controller.IPInPodSubnets(net.ParseIP("10.10.10.10")))
 	assert.Equal(t, false, c.Controller.IPInPodSubnets(net.ParseIP("8.8.8.8")))
 }
+
+func TestRemoveStaleTunnelPorts(t *testing.T) {
+	c, closeFn := newController(t, &config.NetworkConfig{
+		TrafficEncapMode:  0,
+		TunnelType:        ovsconfig.TunnelType("vxlan"),
+		EnableIPSecTunnel: true,
+		IPSecPSK:          "changeme",
+	})
+
+	c.interfaceStore.AddInterface(&interfacestore.InterfaceConfig{
+		Type:          interfacestore.TunnelInterface,
+		InterfaceName: util.GenerateNodeTunnelInterfaceName("xyz-k8s-0-1"),
+		TunnelInterfaceConfig: &interfacestore.TunnelInterfaceConfig{
+			NodeName: "xyz-k8s-0-1",
+			Type:     ovsconfig.TunnelType("vxlan"),
+			PSK:      "mismatchpsk",
+			RemoteIP: nodeIP1,
+		},
+		OVSPortConfig: &interfacestore.OVSPortConfig{
+			PortUUID: "123",
+		},
+	})
+
+	defer closeFn()
+	defer c.queue.ShutDown()
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	c.informerFactory.Start(stopCh)
+	c.informerFactory.WaitForCacheSync(stopCh)
+	node1 := &corev1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "xyz-k8s-0-1",
+		},
+		Spec: corev1.NodeSpec{
+			PodCIDR:  podCIDR.String(),
+			PodCIDRs: []string{podCIDR.String()},
+		},
+		Status: corev1.NodeStatus{
+			Addresses: []corev1.NodeAddress{
+				{
+					Type:    corev1.NodeInternalIP,
+					Address: nodeIP1.String(),
+				},
+			},
+		},
+	}
+
+	c.clientset.CoreV1().Nodes().Create(context.TODO(), node1, metav1.CreateOptions{})
+	c.ovsClient.EXPECT().DeletePort("123").Times(1)
+
+	tests := []struct {
+		name    string
+		wantErr bool
+	}{
+		{
+			name:    "good path with IPSec enabled",
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := c.removeStaleTunnelPorts(); (err != nil) != tt.wantErr {
+				t.Errorf("Controller.removeStaleTunnelPorts() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
```

Comment:
do you think there will be more tests added in the future? maybe since you only have one test right now there is no need to use a table pattern

Reply:
removed table pattern